### PR TITLE
Add configuration property for Hive view security mode

### PR DESCRIPTION
**Pull request by joechu1**
## Description

This PR provides a configuration property to set the security mode used for Hive views. Hive views are hard-coded to use the DEFINER mode, which may not always be the desired mode for checking user permissions.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Hive connector

> How would you describe this change to a non-technical end user or system administrator?

The configuration property determines whether access to a Hive view is granted based on the permissions of the user who created the view, or the user who is querying the view, assuming that the security permission checks are enabled.

## Related issues, pull requests, and links

* Based on the work done in [#7358](https://github.com/trinodb/trino/pull/7358)

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -2420,8 +2423,8 @@ public Optional<ConnectorViewDefinition> getView(ConnectorSession session, Schem
                         throw new HiveViewNotSupportedException(viewName);
                     }
 
-                    ConnectorViewDefinition definition = createViewReader(metastore, session, view, typeManager, this::redirectTable, metadataProvider)
-                            .decodeViewData(view.getViewOriginalText().get(), view, catalogName);
+                    ConnectorViewDefinition definition = createViewReader(metastore, session, view, typeManager, this::redirectTable, metadataProvider, true)
```
Comment (dain) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:2426]: why does this have a hard coded `true` but the next line is configurable?  If this is correct, add a comment explaining what is happening here

Comment (dain) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:2426]: Also, this appears to be the only caller of this in the PR, and it is hard coded to `true`.  Does this needt to be a parameter at all?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java [hunk 2]
```diff
@@ -201,14 +202,14 @@ public ConnectorViewDefinition decodeViewData(String viewSql, Table table, Catal
...
-                        false);
+                        runHiveViewsAsInvoker);
             }
             catch (RuntimeException e) {
                 throw new TrinoException(HIVE_VIEW_TRANSLATION_ERROR,
                         format("Failed to translate Hive view '%s': %s",
                                 table.getSchemaTableName(),
                                 e.getMessage()),
-                        e);
+                                e);
```
Comment (dain) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java:212]: This formatting change doesn't seem right

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 3]
```diff
@@ -1126,6 +1128,19 @@ public boolean isLegacyHiveViewTranslation()
         return this.legacyHiveViewTranslation;
     }
 
+    @Config("hive.run-hive-views-as-invoker")
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:1131]: This is a 3rd config toggle pertaining to Hive Views translation.
I think  we should group the configs together under one namespace: `hive.hive-views.*`.

I posted a PR to discuss that: https://github.com/trinodb/trino/pull/12238

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 4]
```diff
@@ -2420,8 +2423,8 @@ public Optional<ConnectorViewDefinition> getView(ConnectorSession session, Schem
...
+                            .decodeViewData(view.getViewOriginalText().get(), view, catalogName, runHiveViewsAsInvoker);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:2427]: per above, pass the configuration to `createViewReader` and don't pass it to `decodeViewData()`

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 1 above)
Comment (joechu1) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java]: Had the wrong approach; re-wrote this based on findepi's guidance

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java (hunk 2 above)
Comment (joechu1) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java]: Misread this; thought I was fixing the indentation but the original was correct.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java (hunk 3 above)
Comment (joechu1) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java]: I updated the config name to follow what is being proposed in #12238

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 4 above)
Comment (joechu1) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java]: Thanks, re-wrote this based on your guidance.

**Comment by findepi** [comment]
@joechu1 since there is an agreement to rename configs as in https://github.com/trinodb/trino/pull/12238

unfortunately, this will cause conflicts here. can you please rebase your PR on that one, or wait for the other one to be merged?

**Comment by joechu1** [comment]
PR has been rebased

**Comment by findepi** [comment]
(rebased after https://github.com/trinodb/trino/pull/12238 merged)

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java [hunk 5]
```diff
@@ -78,18 +78,19 @@ public static ViewReader createViewReader(
...
         }
         if (isHiveViewsLegacyTranslation(session)) {
-            return new LegacyHiveViewReader();
+            return new LegacyHiveViewReader(runHiveViewRunAsInvoker);
         }
 
         return new HiveViewReader(
                 new CoralSemiTransactionalHiveMSCAdapter(metastore, coralTableRedirectionResolver(session, tableRedirectionResolver, metadataProvider)),
-                typeManager);
+                typeManager, runHiveViewRunAsInvoker);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java:93]: nit: per code style

```suggestion
                typeManager,
                runHiveViewRunAsInvoker);
```

**Review by findepi — APPROVED** [review]

**Comment by mosabua** [comment]
Documentation was missed. I created a PR for adding that in.. please review #12272

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java [hunk 6]
```diff
@@ -223,7 +227,7 @@ public ConnectorViewDefinition decodeViewData(String viewSql, Table table, Catal
                         columns,
                         Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
                         Optional.empty(),
-                        false);
+                        hiveViewsRunAsInvoker);
```
[the remark this answers is not in the document]
Comment (joechu1) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java:230]: This does work; although the ConnectorViewDefinition returns without an owner, that information is then added again in [HiveMetadata#getView](https://github.com/joechu1/trino/blob/f5ac41a8be77b22d2e8027a89decff44892f6099/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L2445).

**Pull request merged**